src/View/ImageLoader.py

```diff
@@ -81,75 +85,82 @@ def load(self, interrupt_flag, progress_callback):
                 return False
 
             progress_callback.emit(("Getting contour data...", 30))
-            dict_raw_contour_data, dict_numpoints = \
-                ImageLoading.get_raw_contour_data(dataset_rtss)
+            dict_raw_contour_data, dict_numpoints = ImageLoading.get_raw_contour_data(
+                dataset_rtss
+            )
 
             # Determine which ROIs are one slice thick
             dict_thickness = ImageLoading.get_thickness_dict(
-                dataset_rtss, read_data_dict)
+                dataset_rtss, read_data_dict
+            )
 
             if interrupt_flag.is_set():  # Stop loading.
                 return False
 
             progress_callback.emit(("Getting pixel LUTs...", 50))
             dict_pixluts = ImageLoading.get_pixluts(read_data_dict)
-
+            progress_callback.emit(("Getting pixel LUTS...", 99))
             if interrupt_flag.is_set():  # Stop loading.
                 return False
 
+            progress_callback.emit(("Loading RTSS...", 1))
             # Add RTSS values to PatientDictContainer
             patient_dict_container.set("rois", rois)
             patient_dict_container.set("raw_contour", dict_raw_contour_data)
             patient_dict_container.set("num_points", dict_numpoints)
             patient_dict_container.set("pixluts", dict_pixluts)
+            progress_callback.emit(("Loading RTSS...", 100))
+            if "rtdose" not in file_names_dict:
+                progress_callback.emit(("Synthesizing RT Dose...", 1))
+                self.load_temp_rtdose(path, progress_callback, interrupt_flag)
+                progress_callback.emit(("Synthesizing RT Dose...", 100))
 
-            if 'rtdose' not in file_names_dict:
-                self.load_temp_rtdose(path,progress_callback,interrupt_flag)
-
-            if 'rtdose' in file_names_dict:
+            if "rtdose" in file_names_dict:
                 # Check to see if DVH data exists in the RT Dose. If
                 # it is there, return (it will be populated later). If
                 # not, ask if the user wants it calculated.
                 try:
+                    progress_callback.emit(("Checking RT Dose for DVH data...", 1))
                     dvh_data = rtdose2dvh()
+                    progress_callback.emit(("Checking RT Dose for DVH data...", 100))
```

Comment:
this goes quickly, have not tested behaviour when DVH is not in RT Dose and user gets asked to calculate.